examples/nftvm/storage/marketplace_storage.go

```diff
@@ -0,0 +1,94 @@
+// Contains read/write logic pertaining to the NFT marketplace
+package storage
+
+import (
+	"context"
+	"encoding/binary"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/state"
+)
+
+const MarketplaceOrderStateChunks = consts.Uint64Len
+
+// TODO: tune this
+const BuyNFTStateChunks = 50
+
+func MarketplaceOrderStateKey(
+	orderID ids.ID,
+) (k []byte) {
+	k = make([]byte, consts.Uint16Len+ids.IDLen+consts.Uint16Len)
+	k[0] = marketplaceOrderPrefix
+	copy(k[1:], orderID[:])
+	binary.BigEndian.PutUint16(k[1+ids.IDLen:], MarketplaceOrderStateChunks)
+	return
+}
+
+func GenerateOrderID(
+	parentCollection codec.Address,
+	instanceNum uint32,
+	price uint64,
+) (ids.ID, error) {
+	v := make([]byte, codec.AddressLen+consts.Uint32Len+consts.Uint64Len)
+	// TODO: Check that this is OK
+	id := hashing.ComputeHash256Array(v)
+	return id, nil
+}
```

Comment:
It looks like this will always construct a byte array of the same length and compute the same hash. The parameters are not used.

Reply:
What should the DB layout for the orders be? Should they be clustered together? If the state key is hashed, then they'll be scattered randomly across the database including interspersing them with other data that's stored under the same prefix.

Reply:
Addressing the first point, I forgot to populate `v` with the function arguments passed through.

Addressing the second point, I choose to hash a portion of order state keys as this way, orders have a negligible probability of colliding with each other. I'm not sure how orders under this model would be organized in the database, but my main thought was that since orders share the same prefix (`marketplaceOrderPrefix`), then they would be together or at least this would allow for iterating across the database for orders. I can definitely change how order IDs are generated though